freud/environment.pyx

```diff
@@ -385,32 +362,29 @@ cdef class LocalDescriptors(Compute):
             l_orientations = orientations
             l_orientations_ptr = <quat[float]*> &l_orientations[0, 0]
 
-        cdef const float[:, ::1] l_points = points
-        cdef unsigned int n_points = l_points.shape[0]
-        cdef const float[:, ::1] l_query_points = query_points
-        cdef unsigned int n_query_points = l_query_points.shape[0]
+        if mode not in self.known_modes:
+            raise RuntimeError(
+                'Unknown LocalDescriptors orientation mode: {}'.format(mode))
         cdef freud._environment.LocalDescriptorOrientation l_mode
-
         l_mode = self.known_modes[mode]
 
         self.thisptr.compute(
-            dereference(b.thisptr), num_neighbors,
-            <vec3[float]*> &l_points[0, 0], n_points,
-            <vec3[float]*> &l_query_points[0, 0], n_query_points,
+            nq.get_ptr(),
+            <vec3[float]*> &l_query_points[0, 0], num_query_points,
             l_orientations_ptr, l_mode,
-            nlist_.get_ptr())
+            nlistptr.get_ptr(), dereference(qargs.thisptr))
         return self
 
+    @Compute._computed_property()
+    def nlist(self):
+        return freud.locality.nlist_from_cnlist(self.thisptr.getNList())
```

Comment:
This method is very similar to the `BoxFromCPP` method. Would be nice to standardize the naming of these somehow. Or just see if they can be class methods of the `Box` and `NeighborList` classes?

Reply:
I'm fine with something like that. I'd prefer doing that in a separate PR since we need to do a lot of renaming.